src/core/pjax.ts

```diff
@@ -363,15 +363,31 @@ class Pjax {
                 const tempDocument: HTMLDocument = document.implementation.createHTMLDocument("pjax-temp-document");
                 tempDocument.documentElement.innerHTML = body;
 
-                let selector = "main";
-                if (request.target !== null) {
-                    selector = `[pjax-id="${request.target}"]`;
+                let selector;
+                if (request.targetSelector !== null) {
+                    selector = `[pjax-id="${request.targetSelector}"]`;
+                } else {
+                    selector = "main";
                 }
 
                 const currentMain = document.body.querySelector(selector);
                 const main = tempDocument.querySelector(selector);
 
                 if (main && currentMain) {
+                    if (main.getAttribute("pjax-id") && currentMain.getAttribute("pjax-id")) {
+                        const incomingId = main
+                            .getAttribute("pjax-id")
+                            .toLowerCase()
+                            .trim();
+                        const currentId = currentMain
+                            .getAttribute("pjax-id")
+                            .toLowerCase()
+                            .trim();
+                        if (incomingId !== currentId) {
+                            console.error("The elements pjax-id attributes don't match.");
```

Comment:
Is this console log going to be showing on production environments?

Reply:
We no longer need to wrap console logs since we switched to [terser](https://www.npmjs.com/package/terser) and enabled the option to `drop_console` during production builds.